dataplicity/m2mmanager.py

```diff
@@ -161,11 +178,17 @@ def manager(self):
         return getattr(self, '_manager', None)
 
     def on_instruction(self, sender, data):
-        self.manager.on_instruction(sender, data)
+        if self.manager is not None:
+            self.manager.on_instruction(sender, data)
 
     def on_close(self, app):
-        super(M2MClient, self).on_close(app)
-        self.manager.on_client_close()
+        if self.manager is not None:
+            super(M2MClient, self).on_close(app)
+            self.manager.on_client_close()
+
+    def abandon(self):
+        self._manager = None
```

Comment:
Un-bind the manager, so there is no chance of some method being called from a hook from a dying thread.